lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Wallet.DB.Sqlite where
+
+import Prelude
+
+import Cardano.Wallet.DB.SqliteTypes
+    ( AddressScheme, TxId )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( UTCTime )
+import Data.Word
+    ( Word32 )
+import Database.Persist.TH
+import GHC.Generics
+    ( Generic (..) )
+
+import qualified Cardano.Wallet.Primitive.Types as W
+
+-- fixme: need tables for wallet AddressPool
+
+share
+    [ mkPersist sqlSettings { mpsPrefixFields = False }
+    , mkMigrate "migrateAll"
+    ]
+    [persistLowerCase|
+
+-- Wallet IDs, address discovery state, and metadata.
+Wallet
+    walTableId                 W.WalletId     sql=wallet_id
+    walTableName               Text           sql=name
+    walTablePassphraseLastUpdatedAt  UTCTime  sql=passphrase_last_updated_at
+    walTableStatus             W.WalletState  sql=status
+    walTableDelegation         Text Maybe     sql=delegation
+    walTableAddressScheme      AddressScheme  sql=address_discovery
+
+    Primary walTableId
+    deriving Show Generic
+
+-- The private key for each wallet. This is in a separate table simply so that
+-- "SELECT * FROM wallet" won't print keys.
+PrivateKey                               sql=private_key
+    privateKeyTableWalletId  W.WalletId  sql=wallet_id
+    privateKeyTableKey       Text        sql=private_key
+
+    Primary privateKeyTableWalletId
+    Foreign Wallet fk_wallet_private_key privateKeyTableWalletId
+
+    deriving Show Generic
+
+-- Maps a transaction ID to its metadata (which is calculated when applying
+-- blocks).
+--
+-- TxMeta is specific to a wallet because multiple wallets may have the same
+-- transaction with different metdata values. The associated inputs and outputs
+-- of the transaction are in the TxIn and TxOut tables.
+TxMeta
+    txMetaTableTxId       TxId         sql=tx_id
+    txMetaTableWalletId   W.WalletId   sql=wallet_id
```

Comment:
I wonder if we really needs here walletId

Reply:
probably yes - if we remove wallet then all pertinent TxMeta should also be removed

Reply:
Ok - my comment above is wrong in many ways. I didn't took into account that `TxMeta` will be different depending on the `Wallet` it bellongs to (as @rvl pointed out)